examples/workload_identity/main.tf

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  cluster_type = "simple-zonal"
+}
+
+provider "google" {
+  version = "~> 2.18.0"
+  region  = var.region
+}
+
+module "gke" {
+  source            = "../../"
+  project_id        = var.project_id
+  name              = "${local.cluster_type}-cluster${var.cluster_name_suffix}"
+  regional          = false
+  region            = var.region
+  zones             = var.zones
+  network           = var.network
+  subnetwork        = var.subnetwork
+  ip_range_pods     = var.ip_range_pods
+  ip_range_services = var.ip_range_services
+  service_account   = "create"
```

Comment:
I had to add identity_namespace = "${module.example-project.project_id}.svc.id.goog" to have workload identity enabled with //modules/beta-public-cluster/. 
Otherwise workload identity fails with  Error 400: Identity namespace does not exist (proj-id.svc.id.goog)